### DISCUSSION TITLE
My project is spitting out multiple files with different hashes, some files are unused by the final bundle.

### DISCUSSION THREAD

**Discussion by kkirby** [parcel-bundler/parcel#7450] [Q&A]
As the long title says, I have a medium sized project that is spitting out multiple files for one source. The files differ in the hash and the content. Some of the files being generated aren't even referenced by the final bundle.

Here's an example.

I have a component called LiveStream. Parcel is generating these files for production:

```
 2995  LiveStream.1d1b0aa6.js
 2960  LiveStream.cfb6c88a.js
22918  LiveStream.d3ad423f.js
```

This is the dependency graph of those three files:

```
LiveStream.1d1b0aa6.js
  frontend.3935dc64.js
    index.html
LiveStream.cfb6c88a.js
  index.8c9a4ba7.js
    index.html
LiveStream.d3ad423f.js
  frontend.379dd93c.js
    -- deadend
```

As you can see `LiveStream.d3ad423f.js` results in a deadend and isn't actually used.

In order to determine the dependency path, I'm using grep on the dist folder. I have a script `find.sh` that takes a hash and finds the files that reference that hash:

```
#!/usr/bin/env bash
set -x
grep -Rl $1 . | grep -v $1 | grep -v \.map
```

Here's the command progression for the last file, `LiveStream.d3ad423f.js`:

<img width="394" alt="CleanShot 2021-12-14 at 16 03 23@2x" src="https://user-images.githubusercontent.com/34333/146086721-16f28ee6-8b6a-4343-8d23-3fd77d505c64.png">

What am I missing? I have a ton of these files that aren't used.

I'm running the latest Parcel, no additional plugins used. React/TS/SCSS.

**Comment by mischnic** [comment]
Do you have async bundles? These might be shared bundles that aren't loaded on page load but later.

**Reply by kkirby** [reply]
I do, yes. But I grepped the dist folder for the file hashes and that's how I determined that some of the files are unused.

**Reply by kkirby** [reply]
I added some info to my post in regards to my method for determining unused files.
